mongodb: 3.4.10 -> 4.0.12

jarjee opened the pull request:
###### Motivation for this change
Bringing the mongodb server up to the latest stable release. Also had some odd issues with the JS interpreter when using "load()", which seems to be fixed in this version.

###### Things done
- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [x] macOS
   - [ ] other Linux distributions
- [x] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

jarjee committed:
mongodb: 3.4.10 -> 4.0.4

fix: Adding libtool to allow darwin compiles

Libtool seems to be required for mongodb to compile on darwin.

fix: Marking MongoDB as broken on aarch64

fix: Adding libtools to the pkg imports

Update mongodb to 4.0.4

jarjee commented:
@GrahamcOfBorg build mongodb

jarjee commented:
cc @offlinehacker @bluescreen303 @wkennington @cstrahan

obadz commented:
@GrahamcOfBorg build mongodb

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=7935b87b-043a-46b2-8121-a4a6b3ffdce5)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
clang++ -o build/nixos/mongo/s/client/shard_remote.o -c -Woverloaded-virtual -Werror=unused-result -Wpessimizing-move -Wredundant-move -Wno-undefined-var-template -Wno-instantiation-after-specialization -std=c++14 -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/bdy0cs6ldrhaqsad7jk0bj5idyb2mh1r-boost-1.60.0/include -I/nix/store/95a2p304va03m7kvj88npmdm8z9qfin2-gperftools-2.6.3/include -I/nix/store/9824k2gmknqp3k4nlcaqpf2pkshva501-pcre-cpp-8.41-bin/include -I/nix/store/qwg3hxrmqk1dcb58l7pnzvr73s9zy7xd-snappy-1.1.7/include -I/nix/store/553hwxk54h8z87m4xgga4n8dznd6n8vp-zlib-1.2.11/include -I/nix/store/sz3q1fs0nvh8x4vnql6js83svi7h0976-libyaml-cpp-0.6.2/include -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/pf16lx06p07zklkdra52p7b5akpd35qj-openssl-1.0.2o-dev/include -I/nix/store/29dslwbdcj62xbip0skfl6hi196jwwii-openssl-1.0.2o/include -I/nix/store/nzp6p9gnyr0764l9w6hp38hvjsndm150-libpcap-1.8.1/include -I/nix/store/n05amix531ayklpwwdrgwlrm7hb2p6z9-python-2.7.14-env/include -I/nix/store/95vh5vrfi7a0cdggl7r5wz29acz11wf5-apple-framework-Security/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-private-field -Wno-deprecated-declarations -Wno-tautological-constant-out-of-range-compare -Wno-unused-const-variable -Wno-missing-braces -Wno-inconsistent-missing-override -Wno-potentially-evaluated-expression -Wno-unused-lambda-capture -Wno-exceptions -fstack-protector-strong -fno-builtin-memcmp -DNDEBUG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Ibuild/nixos -Isrc src/mongo/s/client/shard_remote.cpp
clang++ -o build/nixos/mongo/s/client/version_manager.o -c -Woverloaded-virtual -Werror=unused-result -Wpessimizing-move -Wredundant-move -Wno-undefined-var-template -Wno-instantiation-after-specialization -std=c++14 -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/bdy0cs6ldrhaqsad7jk0bj5idyb2mh1r-boost-1.60.0/include -I/nix/store/95a2p304va03m7kvj88npmdm8z9qfin2-gperftools-2.6.3/include -I/nix/store/9824k2gmknqp3k4nlcaqpf2pkshva501-pcre-cpp-8.41-bin/include -I/nix/store/qwg3hxrmqk1dcb58l7pnzvr73s9zy7xd-snappy-1.1.7/include -I/nix/store/553hwxk54h8z87m4xgga4n8dznd6n8vp-zlib-1.2.11/include -I/nix/store/sz3q1fs0nvh8x4vnql6js83svi7h0976-libyaml-cpp-0.6.2/include -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/pf16lx06p07zklkdra52p7b5akpd35qj-openssl-1.0.2o-dev/include -I/nix/store/29dslwbdcj62xbip0skfl6hi196jwwii-openssl-1.0.2o/include -I/nix/store/nzp6p9gnyr0764l9w6hp38hvjsndm150-libpcap-1.8.1/include -I/nix/store/n05amix531ayklpwwdrgwlrm7hb2p6z9-python-2.7.14-env/include -I/nix/store/95vh5vrfi7a0cdggl7r5wz29acz11wf5-apple-framework-Security/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-private-field -Wno-deprecated-declarations -Wno-tautological-constant-out-of-range-compare -Wno-unused-const-variable -Wno-missing-braces -Wno-inconsistent-missing-override -Wno-potentially-evaluated-expression -Wno-unused-lambda-capture -Wno-exceptions -fstack-protector-strong -fno-builtin-memcmp -DNDEBUG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Ibuild/nixos -Isrc src/mongo/s/client/version_manager.cpp
clang++ -o build/nixos/mongo/db/startup_warnings_common.o -c -Woverloaded-virtual -Werror=unused-result -Wpessimizing-move -Wredundant-move -Wno-undefined-var-template -Wno-instantiation-after-specialization -std=c++14 -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/bdy0cs6ldrhaqsad7jk0bj5idyb2mh1r-boost-1.60.0/include -I/nix/store/95a2p304va03m7kvj88npmdm8z9qfin2-gperftools-2.6.3/include -I/nix/store/9824k2gmknqp3k4nlcaqpf2pkshva501-pcre-cpp-8.41-bin/include -I/nix/store/qwg3hxrmqk1dcb58l7pnzvr73s9zy7xd-snappy-1.1.7/include -I/nix/store/553hwxk54h8z87m4xgga4n8dznd6n8vp-zlib-1.2.11/include -I/nix/store/sz3q1fs0nvh8x4vnql6js83svi7h0976-libyaml-cpp-0.6.2/include -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/pf16lx06p07zklkdra52p7b5akpd35qj-openssl-1.0.2o-dev/include -I/nix/store/29dslwbdcj62xbip0skfl6hi196jwwii-openssl-1.0.2o/include -I/nix/store/nzp6p9gnyr0764l9w6hp38hvjsndm150-libpcap-1.8.1/include -I/nix/store/n05amix531ayklpwwdrgwlrm7hb2p6z9-python-2.7.14-env/include -I/nix/store/95vh5vrfi7a0cdggl7r5wz29acz11wf5-apple-framework-Security/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-private-field -Wno-deprecated-declarations -Wno-tautological-constant-out-of-range-compare -Wno-unused-const-variable -Wno-missing-braces -Wno-inconsistent-missing-override -Wno-potentially-evaluated-expression -Wno-unused-lambda-capture -Wno-exceptions -fstack-protector-strong -fno-builtin-memcmp -DNDEBUG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/startup_warnings_common.cpp
libtool -static -o build/nixos/mongo/db/libstartup_warnings_common.a -s -no_warning_for_no_symbols build/nixos/mongo/db/startup_warnings_common.o
sh: libtool: command not found
scons: *** [build/nixos/mongo/db/libstartup_warnings_common.a] Error 127
scons: building terminated because of errors.
build/nixos/mongo/db/libstartup_warnings_common.a failed: Error 127
builder for '/nix/store/5yxbq91zjhp7nmky38a8brcp3sa9sq6y-mongodb-3.6.4.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/5yxbq91zjhp7nmky38a8brcp3sa9sq6y-mongodb-3.6.4.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=beeee4cc-3c1e-42c3-945c-92db63f0b8b4)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
build/nixos/mongo/db/commands/authentication_commands.o failed: Error 4
build/nixos/mongo/db/commands/parameters.o failed: Error 4
build/nixos/mongo/db/commands/fail_point_cmd.o failed: Error 4
build/nixos/mongo/db/commands/find_and_modify_common.o failed: Error 4
build/nixos/mongo/s/mongos_options.o failed: Error 4
build/nixos/mongo/db/commands/hashcmd.o failed: Error 4
build/nixos/mongo/db/commands/isself.o failed: Error 4
build/nixos/mongo/db/commands/conn_pool_sync.o failed: Error 4
builder for '/nix/store/h1brhkx7knhd0mp33mm9fczzhhyhj3yp-mongodb-3.6.4.drv' failed with exit code 2
[31;1merror:[0m build of '/nix/store/h1brhkx7knhd0mp33mm9fczzhhyhj3yp-mongodb-3.6.4.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=ca9331c6-e526-464f-8d51-ebca356b05a2)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4
shrinking /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4/bin/mongo
shrinking /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4/bin/mongod
shrinking /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4/bin/mongos
shrinking /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4/bin/mongoperf
strip is /nix/store/j75dgadrff2d1fyc4fczmcgqkid2imdx-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4/lib  /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4/bin
patching script interpreter paths in /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4
checking for references to /build in /nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4...
/nix/store/9qq8y8vv5vbblllbvgv2bngzmd1qxq4m-mongodb-3.6.4
```
</p></details>

obadz reviewed (changes requested):
Needs darwin fix

obadz reviewed (changes requested):
Also needs aarch64 to be marked broken (or fixed)

jarjee commented:
I don't have access to a mac to test with, but adding `libtools` should do it.

jarjee commented:
@GrahamcOfBorg build mongodb

obadz commented:
@GrahamcOfBorg build mongodb

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=90de8b5f-32e0-4f56-8389-45f392736537)

The following builds were skipped because they don't evaluate on aarch64-linux: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=e68c8d2d-f956-4c9e-80f7-aad2d31ddb6f)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
clang++ -o build/nixos/mongo/db/cloner.o -c -Woverloaded-virtual -Werror=unused-result -Wpessimizing-move -Wredundant-move -Wno-undefined-var-template -Wno-instantiation-after-specialization -std=c++14 -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/bdy0cs6ldrhaqsad7jk0bj5idyb2mh1r-boost-1.60.0/include -I/nix/store/95a2p304va03m7kvj88npmdm8z9qfin2-gperftools-2.6.3/include -I/nix/store/9824k2gmknqp3k4nlcaqpf2pkshva501-pcre-cpp-8.41-bin/include -I/nix/store/qwg3hxrmqk1dcb58l7pnzvr73s9zy7xd-snappy-1.1.7/include -I/nix/store/553hwxk54h8z87m4xgga4n8dznd6n8vp-zlib-1.2.11/include -I/nix/store/sz3q1fs0nvh8x4vnql6js83svi7h0976-libyaml-cpp-0.6.2/include -I/nix/store/wrpcwr7k2x6r0s5xc1svvpsha0cclxjk-cyrus-sasl-2.1.26-bin/include -I/nix/store/pf16lx06p07zklkdra52p7b5akpd35qj-openssl-1.0.2o-dev/include -I/nix/store/29dslwbdcj62xbip0skfl6hi196jwwii-openssl-1.0.2o/include -I/nix/store/nzp6p9gnyr0764l9w6hp38hvjsndm150-libpcap-1.8.1/include -I/nix/store/n05amix531ayklpwwdrgwlrm7hb2p6z9-python-2.7.14-env/include -I/nix/store/95vh5vrfi7a0cdggl7r5wz29acz11wf5-apple-framework-Security/include -I/nix/store/a4iyj890mzb1n1zd5cyxgyd3b9z3gqrk-libtool-2.4.6/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-unused-private-field -Wno-deprecated-declarations -Wno-tautological-constant-out-of-range-compare -Wno-unused-const-variable -Wno-missing-braces -Wno-inconsistent-missing-override -Wno-potentially-evaluated-expression -Wno-unused-lambda-capture -Wno-exceptions -fstack-protector-strong -fno-builtin-memcmp -DNDEBUG -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/cloner.cpp
libtool -static -o build/nixos/mongo/db/bson/libdotted_path_support.a -s -no_warning_for_no_symbols build/nixos/mongo/db/bson/dotted_path_support.o
Usage: /nix/store/a4iyj890mzb1n1zd5cyxgyd3b9z3gqrk-libtool-2.4.6/bin/libtool [OPTION]... [MODE-ARG]...
Try 'libtool --help' for more information.
libtool:   error: unrecognised option: '-static'
scons: *** [build/nixos/mongo/db/bson/libdotted_path_support.a] Error 1
scons: building terminated because of errors.
build/nixos/mongo/db/bson/libdotted_path_support.a failed: Error 1
builder for '/nix/store/s412p7bdjj0zjrnsc58jbdf4pcgf9sf8-mongodb-3.6.4.drv' failed with exit code 2
error: build of '/nix/store/s412p7bdjj0zjrnsc58jbdf4pcgf9sf8-mongodb-3.6.4.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=005af917-0ef8-4a19-85d7-63df09bbc276)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
g++ -o build/nixos/mongo/db/s/config/configsvr_create_database_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/config/configsvr_create_database_command.cpp
g++ -o build/nixos/mongo/db/s/merge_chunks_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/merge_chunks_command.cpp
g++ -o build/nixos/mongo/db/s/set_shard_version_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/set_shard_version_command.cpp
g++ -o build/nixos/mongo/db/s/config/configsvr_split_chunk_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/config/configsvr_split_chunk_command.cpp
g++ -o build/nixos/mongo/db/s/check_sharding_index_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/check_sharding_index_command.cpp
g++ -o build/nixos/mongo/db/s/get_shard_version_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/get_shard_version_command.cpp
g++ -o build/nixos/mongo/db/s/config/configsvr_enable_sharding_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/config/configsvr_enable_sharding_command.cpp
g++ -o build/nixos/mongo/db/s/config/configsvr_update_zone_key_range_command.o -c -Woverloaded-virtual -Wno-maybe-uninitialized -std=c++14 -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/jxcxnmvvh6lmwlrgql44fpxlrvnknzbf-boost-1.60.0/include -I/nix/store/kqlavdmnqhnd9pnviwhji8762gfkhf1d-gperftools-2.6.3/include -I/nix/store/za04kj9q5m08bh946x96y6cip1kchn9c-pcre-cpp-8.41-bin/include -I/nix/store/grffx916ray3d411wf9czx2y127jac6g-snappy-1.1.7/include -I/nix/store/vj9716h62y21a2hv18al8zvhlqpl30sx-zlib-1.2.11/include -I/nix/store/c85m7cxs10m0rfrqywdr4dly8dnp2gmm-libyaml-cpp-0.6.2/include -I/nix/store/mhrn4ygm88n8x4kb1myrnff41x690r2a-cyrus-sasl-2.1.26-bin/include -I/nix/store/vp2imwdhvqfi2l2d39zqhqz7y56aw2kl-openssl-1.0.2o-dev/include -I/nix/store/qahj8wb88rfs00l9xmjg74m6c52lnxiy-openssl-1.0.2o/include -I/nix/store/hy5jf061b9wxws6wijn2qjcraa569w0l-libpcap-1.8.1/include -I/nix/store/8k3xn58nw9b72hfnrnvjb03v3njx8akq-python-2.7.14-env/include -fno-omit-frame-pointer -fno-strict-aliasing -ggdb -pthread -Wall -Wsign-compare -Wno-unknown-pragmas -Winvalid-pch -O2 -Wno-unused-local-typedefs -Wno-unused-function -Wno-deprecated-declarations -Wno-unused-const-variable -Wno-unused-but-set-variable -Wno-missing-braces -fstack-protector-strong -fno-builtin-memcmp -fPIE -DNDEBUG -D_FORTIFY_SOURCE=2 -DBOOST_SYSTEM_NO_DEPRECATED -DBOOST_MATH_NO_LONG_DOUBLE_MATH_FUNCTIONS -Isrc/third_party/s2 -Ibuild/nixos -Isrc src/mongo/db/s/config/configsvr_update_zone_key_range_command.cpp
building of '/nix/store/2nnpsdk2cralpp249dixs74xlw9f374a-mongodb-3.6.4.drv' timed out after 1800 seconds
error: build of '/nix/store/2nnpsdk2cralpp249dixs74xlw9f374a-mongodb-3.6.4.drv' failed
```
</p></details>

xeji commented:
@GrahamcOfBorg  build mongodb

GrahamcOfBorg commented:
No attempt on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=efa43510-399b-492a-822c-253021122f81)

The following builds were skipped because they don't evaluate on aarch64-linux: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```

a) For `nixos-rebuild` you can set
  { nixpkgs.config.allowBroken = true; }
in configuration.nix to override this.

b) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
  { allowBroken = true; }
to ~/.config/nixpkgs/config.nix.


```
</p></details>

GrahamcOfBorg commented:
Failure on x86_64-darwin [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=aa766f8a-ef96-4eba-9114-d17d67881d59)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
Usage: /nix/store/v19mh1p18achhbcb3r4mf1an2ymyd654-libtool-2.4.6/bin/libtool [OPTION]... [MODE-ARG]...
scons: *** [build/nixos/third_party/wiredtiger/libwiredtiger.a] Error 1
Try 'libtool --help' for more information.
libtool:   error: unrecognised option: '-static'
scons: *** [build/nixos/third_party/s2/base/libbase.a] Error 1
scons: building terminated because of errors.
build/nixos/third_party/wiredtiger/libwiredtiger.a failed: Error 1
build/nixos/third_party/s2/base/libbase.a failed: Error 1
builder for '/nix/store/nxnpb4mg5hig1xpryannvi9yy5sf731k-mongodb-3.6.4.drv' failed with exit code 2
error: build of '/nix/store/nxnpb4mg5hig1xpryannvi9yy5sf731k-mongodb-3.6.4.drv' failed
```
</p></details>

GrahamcOfBorg commented:
Success on x86_64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.40082&attempt_id=17e11827-8e1b-4043-932f-cf26e0a2b861)

Attempted: mongodb

<details><summary>Partial log (click to expand)</summary><p>

```
shrinking RPATHs of ELF executables and libraries in /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4
shrinking /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4/bin/mongod
shrinking /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4/bin/mongo
shrinking /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4/bin/mongos
shrinking /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4/bin/mongoperf
strip is /nix/store/21ymadblbmsbb2bk4q7gl4kjasp8zmgd-binutils-2.30/bin/strip
stripping (with command strip and flags -S) in /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4/lib  /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4/bin
patching script interpreter paths in /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4
checking for references to /build in /nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4...
/nix/store/ghw2q6a1pdqzv46wqcnydqylh7n04f55-mongodb-3.6.4
```
</p></details>

cyounkins commented:
Hello! @jarjee could you rebase this, or may I make a new PR with your changes?

Also, Mongo is no longer supported on 32-bit systems - "Changed in version 3.4: MongoDB no longer supports 32-bit x86 platforms." so I think `meta.platforms` should be set to `stdenv.lib.subtractLists platforms.doubles.i686 platforms.doubles.unix;` (Reference #52463)

cyounkins commented:
It appears the 32-bit issue remains in your most recent push - We need to disable building on 32-bit systems as it isn't supported and in my testing has failed. This is the error that occurs: https://jira.mongodb.org/browse/SERVER-22810

Could you please set `meta.platforms` to `stdenv.lib.subtractLists platforms.doubles.i686 platforms.doubles.unix;` ? Thanks!

jarjee commented:
@cyounkins Sorry for the wait, I've made the change you requested.

ryantm commented:
@GrahamcOfBorg build mongodb

schmittlauch commented on lib/licenses.nix:
```diff
@@ -624,6 +624,12 @@ lib.mapAttrs (n: v: v // { shortName = n; }) rec {
+    shortName = "SSPL";
+    fullName = "Server Side Public License";
+    url = https://www.mongodb.com/licensing/server-side-public-license;
+  };
```
This lacks a `free = false;`, as the SSPL is not recognized by the OSI and also has been [dropped from Fedora for that reason.](https://www.geekwire.com/2019/mongodbs-licensing-changes-led-red-hat-drop-database-latest-version-server-os/)

dpausp commented:
I'm not an expert for MongoDB, but the docs say that upgrades to 4.0 can only be done from an existing 3.6 installation. 3.4 should be left as default and 3.6 and 4.0 should be provided as additional versions to make upgrades possible, I think.

aanderse commented:
@jarjee and/or @cyounkins are you able to address the above mentioned issue(s) and continue with this PR?

aanderse commented:
@jarjee great, thanks! Let us know if you have any issues. Looking forward to getting this merged!

jarjee committed:
mongodb: split packages to expose 3.4, 3.6 and 4.0

jarjee commented:
I've just noticed #35298, so I'll also have a look and see if we need any of those patches.

Since we're also adding new packages, I'm thinking we'll also need to look at updating the tests so we test the different mongodb versions as well.

aanderse commented:
@jarjee Yeah that sounds good. If you get stuck at any point please do ping this thread for help as I believe there are a number of people here with motivation to see this through. Thanks again for your work on this.

nixos-discourse commented:
This pull request has been mentioned on **Nix community**. There might be relevant details there:

https://discourse.nixos.org/t/prs-in-distress/3604/1

jarjee committed:
mongodb: 3.4.20 -> 3.4.22, 3.6.12 -> 3.6.13, 4.0.9 -> 4.0.11

jarjee committed:
nixos/tests/mongodb: test against mongodb versions 3.4, 3.6, 4.0

Now has tests for 3.4, 3.6, 4.0. Has some duplication, but it appears to
work on my machine.

aanderse commented:
Thanks @jarjee. What is the status on this now? Are you waiting on anyone/anything? Freeze for `19.09` has been announced for September 7th.

jarjee commented:
Hi @aanderse Only issue that I currently have now is that this doesn't build on OSX. I just got access to a mac with limited access, so I've been trying to debug the build issues there.

aanderse commented:
@GrahamcOfBorg build mongodb-3_4 mongodb-3_6 mongodb-4_0

aanderse commented:
@jarjee you are awesome! Does this work on `darwin` now, do you need some assistance, etc...?

Thank you for your amazing work!

jarjee commented:
@aanderse I'm still fiddling with 4.0, but 3.4 & 3.6 seem to be building now

alyssais commented on pkgs/servers/nosql/mongodb/mongodb.nix:
```diff
@@ -97,9 +105,10 @@ in stdenv.mkDerivation rec {
     description = "A scalable, high-performance, open source NoSQL database";
     homepage = http://www.mongodb.org;
-    license = licenses.agpl3;
+    license = licenses.sspl;
```
Won’t this mean that AGPL versions of MongoDB are marked as non-free?

jarjee replied:
Good question, according to the [MongoDB README](https://github.com/mongodb/mongo/blob/b73a7d6ab9d38382a6c2f1fd0b5c8369d54d8b95/README#L89):

```
LICENSE

  MongoDB is free and the source is available. Versions released prior to
  October 16, 2018 are published under the AGPL. All versions released after
  October 16, 2018, including patch fixes for prior versions, are published
  under the Server Side Public License (SSPL) v1. See individual files for
  details.
```

Versions [3.4.22](https://docs.mongodb.com/manual/release-notes/3.4/#aug-6-2019), [3.6.13](https://docs.mongodb.com/manual/release-notes/3.6/#jun-10-2019) and [4.0.11](https://docs.mongodb.com/manual/release-notes/4.0/#jul-26-2019) were all released after October 16, 2018; so they all have to be marked as SSPL.

alyssais replied:
In that case, I’d be inclined to either not offer patch versions after that date, or to at least continue to offer the last free version.

jarjee commented:
Current state of this PR: I have 4.0, 3.6 and 3.4 building on OSX now :tada:. They don't seem to build in sandbox mode, I'm not sure if this is a problem / how to resolve this.

The other things that I'd like to do:
- [ ] Update v4_0 to use 4.0.12
- [ ] Clean up the nix functions, since there are variables that I'm pretty sure aren't actually being used in the builds

offlinehacker commented on nixos/tests/mongodb.nix:
```diff
@@ -12,29 +12,76 @@ in {
-    startAll;
-    $one->waitForUnit("mongodb.service");
-    $one->succeed("mongo -u nixtest -p nixtest nixtest ${testQuery}") =~ /hello/ or die;
+         $v3_4->start;
```
I think it would be better to have generic testScript, that's version independent.

offlinehacker reviewed (commented):
Why are you using multiple nodes in what is basically a single test with different parameters? I would much rather prefer running a single node for single mongodb version and split into multiple tests.

aanderse commented:
From my perspective we're almost out of time on this. The freeze is coming and this is *very* important to merge. Thank you **so** much @jarjee for your work on this! I'm planning on merging this in about a weeks time. Hopefully everything at that point will be acceptable to all parties involved. @offlinehacker if the tests are modified to your specification will you be satisfied?

jarjee committed:
mongodb-4_0: 4.0.11 -> 4.0.12

jarjee commented:
So the most recent openssl update seems to have broken 3_4, but I've now moved all the tests to use a single machine in 211f84c, based off the feedback from @offlinehacker.

jarjee commented:
I've tried using the old version of openssl with 3_4, but it still doesn't seem to build. Looks like somehow the newer version of openssl is still getting included, even with passing down the old version explicitly.

jarjee commented:
Is it worth adding AGPL variants? Essentially what  @alyssais suggested? I was inclined to not do that since it means that the packages will no longer update, but SSPL being non-free is causing issues. Especially since hydra doesn't run unfree tests AFAIK.

aanderse commented:
@jarjee if only `3.4` is broken then maybe it is worth adding AGPL variant for that version alone. We're so close here...

alyssais commented:
> @jarjee if only `3.4` is broken then maybe it is worth adding AGPL variant for that version alone. We're so close here...

I think 4.1, being the last release series to be available as AGPL,
should have an AGPL variant available in Nixpkgs.

Commit:
mongodb: 3.4.22 -> 3.4.24 & fix ssl

Reverts previous ssl fix commit and updates the mongodb version which fixes the ssl compile problem on darwin.

tricktron commented:
I fixed the ssl error of `v.3.4` in #81027. How could I have added the commits to this pr instead of creating a new one?

jarjee commented:
Hi @tricktron sorry for the wait, thanks for making the fix. I can cherry pick your PRs across if that's ok with you.

If you wanted to add these commits to this PR instead of making a new one, you'd need to add them to my forked version of nixpkgs (adding the commits to [`jarjee/nixpkgs:update/mongodb`](https://github.com/jarjee/nixpkgs/tree/update/mongodb))

tricktron commented:
> Hi @tricktron sorry for the wait, thanks for making the fix. I can cherry pick your PRs across if that's ok with you.

Yeah, that is fine!

> If you wanted to add these commits to this PR instead of making a new one, you'd need to add them to my forked version of nixpkgs (adding the commits to [`jarjee/nixpkgs:update/mongodb`](https://github.com/jarjee/nixpkgs/tree/update/mongodb))

Makes sense, thanks for the info.

uri-canva commented:
Tested this on macOS and it works.

dasJ commented:
This is proably more important than expected (also for backport to 20.03) because the mongodb package does not build with:
```
in job ‘nixos.tests.mongodb.x86_64-linux’:
Package ‘openssl-1.0.2u’ in /nix/store/76gxrrf0jmapcvw25prlqi6m89aa8q3p-source/pkgs/development/libraries/openssl/default.nix:130 is marked as insecure, refusing to evaluate.


Known issues:
 - Support for OpenSSL 1.0.2 ended with 2019.

You can install it anyway by whitelisting this package, using the
following methods:
```
The MongoDB package on both master and 20.03 does not build with the newer OpenSSL 1.1
As NixOS should probably keep support for MongoDB, this needs to be both merged and backported.

Ma27 committed:
mongodb: builds on aarch64 as well

Ma27 reviewed (approved):
LGTM. As @dasJ already mentioned, this should go into 20.03, otherwise we'll only ship unsecure versions of mongodb which isn't a good idea IMHO.

I just tested `mongodb` with the aarch64 commnunity builder and it appears to work fine, so unmarked it as broken.

Are there any more things that would keep us from merging this PR?

@GrahamcOfBorg test mongodb

jul1u5 commented:
Ping. This would be very useful. Can we merge this?

aanderse commented:
I would love that. I lost track where we're at with this, though. Are you running with this now @Ma27?

Ma27 commented:
> Are you running with this now

In `#nixos-dev` everybody seemed to be fine with this. Will merge (and backport mongodb-3.4.24) tomorrow :)

Ma27 committed:
nixos/tests/mongodb: rewrite with python

perl-based VM tests are deprecated.

Ma27 committed:
nixos/release-notes: mention mongodb update

Ma27 commented:
I rewrote the test in python, rebased to master and cleaned up the history. As soon as the mongodb builds (and test) I'm running locally are finished, I'll push the changeset and merge :)

Ma27 commented:
@GrahamcOfBorg test mongodb

Ma27 commented:
@GrahamcOfBorg build mongodb-4_0 mongodb-3_4 mongodb-3_6

Ma27 commented:
@jarjee @worldofpeace @offlinehacker @aanderse can you please take a look at the license (sspl)?

According to MongoDB's FAQ it's free, based on GPLv3, but not (yet?) approved by the OSI (https://www.mongodb.com/licensing/server-side-public-license/faq).

From my understanding it should be therefore possible to mark `licenses.ssdp` as free, however I'm definetely not a software licensing expert or a lawyer. Or do we only mark OSI-approved licenses as free? (Didn't find any reference about this while skimming through the manual).

The main reason I'm asking is because it takes quite some time and resources to compile `mongodb` and the unfree licenses would mean that everyone has to compile it locally.

I guess one of the most important paragraphs from the FAQ is the following:

> Although the SSPL is not OSI approved, it maintains all of the same freedoms the community has always had with MongoDB under AGPL. Users are free to review, modify, and distribute the software or redistribute modifications to the software. However, the Open Source Initiative (OSI) has its own process for approving what it considers to be an open source license, and the SSPL has not received OSI approval. MongoDB software licensed under the SSPL is not considered open source by the OSI.

Ma27 commented:
Merging for now. As discussed in [`#nixos-dev`](https://logs.nix.samueldr.com/nixos-dev/2020-03-26#3227422;), this is at least an improvement over the current state. We need to find a solution for the licensing issue soon though.

adrian-gierakowski commented:
Thanks for this! Any idea if it would take much work to also add mongo 4.2?

Ma27 commented:
Without having it tested: I don't think so, the packages are already splitted, so it should be fairly easy to add another one :)

For the record: I didn't forget the backport, I'll file a PR for this tomorrow hoping that I get a bit more feedback on #83433 first :)
